external/anomaly/tests/test_ote_anomaly_segmentation_training.py

```diff
@@ -0,0 +1,420 @@
+from ote_sdk.test_suite.e2e_test_system import DataCollector, e2e_pytest_performance
+from ote_sdk.test_suite.training_tests_common import (make_path_be_abs,
+                                                      make_paths_be_abs,
+                                                      performance_to_score_name_value,
+                                                      KEEP_CONFIG_FIELD_VALUE,
+                                                      REALLIFE_USECASE_CONSTANT,
+                                                      ROOT_PATH_KEY)
+from ote_sdk.test_suite.training_tests_helper import (OTETestHelper,
+                                                      DefaultOTETestCreationParametersInterface,
+                                                      OTETrainingTestInterface)
+from ote_sdk.test_suite.training_tests_actions import (create_environment_and_task,
+                                                       OTETestTrainingAction,
+                                                       BaseOTETestAction,
+                                                       OTETestTrainingEvaluationAction,
+                                                       OTETestExportAction,
+                                                       OTETestExportEvaluationAction,
+                                                       OTETestPotAction,
+                                                       OTETestPotEvaluationAction,
+                                                       OTETestNNCFAction,
+                                                       OTETestNNCFEvaluationAction,
+                                                       OTETestNNCFExportAction,
+                                                       OTETestNNCFExportEvaluationAction,
+                                                       OTETestNNCFGraphAction)
+
+logger = logging.getLogger(__name__)
+
+@pytest.fixture
+def ote_test_domain_fx():
+    return 'anomaly-segmentation'
+
+@pytest.fixture(scope='session')
+def ote_templates_root_dir_fx():
+    import os.path as osp
+    import logging
+    logger = logging.getLogger(__name__)
+    root = osp.dirname(osp.dirname(osp.realpath(__file__)))
+    root = f'{root}/anomaly_segmentation/configs/'
+    logger.debug(f'overloaded ote_templates_root_dir_fx: return {root}')
+    return root
+
+def DATASET_PARAMETERS_FIELDS() -> List[str]:
+    return deepcopy(['dataset_path',
+                     ])
+
+DatasetParameters = namedtuple('DatasetParameters', DATASET_PARAMETERS_FIELDS())
+
+
+def _get_dataset_params_from_dataset_definitions(dataset_definitions, dataset_name):
+    if dataset_name not in dataset_definitions:
+        raise ValueError(f'dataset {dataset_name} is absent in dataset_definitions, '
+                         f'dataset_definitions.keys={list(dataset_definitions.keys())}')
+    cur_dataset_definition = dataset_definitions[dataset_name]
```

Comment:
I'm having trouble understanding what `cur` means in `cur_dataset_definition`? Would it be possible to rename the variable without abbreviations?

Reply:
renamed